### PR TITLE
Fix warnings which is occuring during compilation of babel 

### PR DESCRIPTION
**Pull request by sumitj824**
### Description

So far we were  getting lots of warning during compilation of babel. This commit fix warnings which were occurring during compilation of babel and add flags so that warnings will be treated as errors. 
 
Engine PR : https://github.com/babelfish-for-postgresql/postgresql_modified_for_babelfish/pull/71
### Issues Resolved

[BABEL-3666]

Signed-off-by: Sumit Jaiswal <sumiji@amazon.com>

### Testing
We don't need to add more tests as we already have test coverage for functions which are getting modified in this commit to fix the warnings in them. Also these changes has been verified by the corresponding author.
* **Use case based -**
 NA

* **Boundary conditions -**
  NA

* **Arbitrary inputs -**
  NA

* **Negative test cases -**
  NA

* **Minor version upgrade tests -**
 NA

* **Major version upgrade tests -**
 NA

* **Performance tests -**
  NA

* **Tooling impact -**
   NA

* **Client tests -**
  JDBC, ODBC, .NET, Python drivers

### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: contrib/babelfishpg_tsql/src/plerrcodes.h [hunk 1]
```diff
@@ -348,6 +348,10 @@
 	"sql_json_scalar_required", ERRCODE_SQL_JSON_SCALAR_REQUIRED
 },
 
+{
+	"sql_json_item_cannot_be_cast_to_target_type", ERRCODE_SQL_JSON_ITEM_CANNOT_BE_CAST_TO_TARGET_TYPE
+},
+
```
Comment (Deepesh125) [inline contrib/babelfishpg_tsql/src/plerrcodes.h]: check this diff

File: contrib/babelfishpg_tsql/src/pltsql_ruleutils.c [hunk 2]
```diff
@@ -463,20 +473,22 @@ tsql_get_functiondef(PG_FUNCTION_ARGS)
...
-	
+ 
+	/* Emit some miscellaneous options on one line */
 	if (proc->proisstrict)
 		appendStringInfoString(&buf, " WITH RETURNS NULL ON NULL INPUT");
 
 	/* And finally the function definition ... */
 	(void) SysCacheGetAttr(PROCOID, proctup, Anum_pg_proc_prosqlbody, &isnull);
-
+
```
Comment (Deepesh125) [inline contrib/babelfishpg_tsql/src/pltsql_ruleutils.c]: remove this space

File: contrib/babelfishpg_tsql/src/pltsql_ruleutils.c [hunk 3]
```diff
@@ -438,22 +439,31 @@ tsql_get_functiondef(PG_FUNCTION_ARGS)
 		appendStringInfoString(&buf, "(");
 	
 	/* we will not pfree name because as we can see name = NameStr(proc->proname) 
-         * here we are not allocating extra space for name, we’re just using proc-> proname.
-         * also at the end, we’re releasing proctup (that will free proc->proname).  
-         */
-	pfree(nsp);
+	 * here we are not allocating extra space for name, we’re just using proc-> proname.
+	 * also at the end, we’re releasing proctup (that will free proc->proname).  
+	 */
```
Comment (Deepesh125) [inline contrib/babelfishpg_tsql/src/pltsql_ruleutils.c:444]: remove extra spaces

**Review by Deepesh125 — CHANGES_REQUESTED** [review]

File: contrib/babelfishpg_tsql/src/plerrcodes.h (hunk 1 above)
Comment (sumitj824) [reply contrib/babelfishpg_tsql/src/plerrcodes.h]: removed

File: contrib/babelfishpg_tsql/src/pltsql_ruleutils.c (hunk 2 above)
Comment (sumitj824) [reply contrib/babelfishpg_tsql/src/pltsql_ruleutils.c]: removed

File: contrib/babelfishpg_tsql/src/pltsql_ruleutils.c (hunk 3 above)
Comment (sumitj824) [reply contrib/babelfishpg_tsql/src/pltsql_ruleutils.c:444]: there isn't any extra spaces

**Review by Deepesh125 — APPROVED** [review]

**Pull request merged**